table-api/src/main/java/io/deephaven/api/ConcurrencyControl.java

```diff
@@ -36,4 +36,30 @@ public interface ConcurrencyControl<T> {
      * @return a new instance of T with serial concurrency control applied.
      */
     T withSerial();
+
+    /**
+     * Designates the expression as a barrier filter with the specified barrier name.
```

Comment:
If we want to use the ConcurrencyControl for select as well; then we should not write this Javadoc in terms of only filters.